### PR TITLE
ARGO-209 Wrong key should return 401 unauthorized

### PR DESCRIPTION
**Pull request by pkoro**
please review /cc @makistsantekidis @skanct

### CODE REVIEW

File: app/results/endpointgroup_test.go [hunk 1]
```diff
@@ -317,6 +317,17 @@ func (suite *endpointGroupAvailabilityTestSuite) TestListEndpointGroupAvailabili
...
 
+	request, _ = http.NewRequest("GET", "/api/v2/results/Report_A/SITE/ST01?start_time=2015-06-20T12:00:00Z&end_time=2015-06-23T23:00:00Z", strings.NewReader(""))
+	request.Header.Set("x-api-key", "AWRONGKEY")
+	request.Header.Set("Accept", "application/xml")
+
+	response = httptest.NewRecorder()
+
+	suite.router.ServeHTTP(response, request)
+
+	// Check that we must have a 200 ok code
```
Comment (skanct) [inline app/results/endpointgroup_test.go]: Comment should say 401

**Comment by pkoro** [comment]
please review /cc @kaggis @dpavlos

File: app/results/routing.go [hunk 2]
```diff
@@ -93,6 +93,11 @@ func routeGroup(r *http.Request, cfg config.Config) (int, http.Header, []byte, e
 	vars := mux.Vars(r)
 	tenantcfg, err := authentication.AuthenticateTenant(r.Header, cfg)
 	if err != nil {
+		if err.Error() == "Unauthorized" {
+			code = http.StatusUnauthorized
+			return code, h, output, err
```
Comment (dpavlos) [inline app/results/routing.go]: Should we also return an error message in the response or just the error code? In such a case we have to also return the output var with an appropriate value.

**Comment by pkoro** [comment]
please review /cc @dpavlos

**Comment by pkoro** [comment]
fixed series of actions in controller. please review /cc @dpavlos

**Comment by dpavlos** [comment]
+1

**Pull request merged**